### DISCUSSION TITLE
Push to origin failed

### DISCUSSION THREAD

**Discussion by Gummybear405** [community/community#24026] [New to GitHub]
<p>Hello, I’m trying to push some code I made to origin, but Im greeted by an error message that follows: error: cannot spawn C:\Users&lt;confidential&gt;\AppData\Local\GitHubDesktop\app-2.6.3\resources\app\static\ask-pass-trampoline.bat: Permission denied<br>
fatal: could not read Username for ‘<a href="https://github.com" rel="noopener nofollow ugc">https://github.com</a>’: terminal prompts disabled. Can anyone help me and tell me what to do?</p>

**Comment by byrneh** [comment, marked as answer]
<p>Hi <a class="mention" href="/gummybear405">@Gummybear405</a>, welcome to the community</p>
<p>I experienced and investigated what is likely to be the same problem on a corporate Win 10 device with some stricter group security policy settings configured when attempting a git clone of a <strong>private</strong> repository , as it calls the same ‘bat’ file to determine username credentials (not a problem for clonng public as authentication is not required).<br>
It worked without problem on a personal Win 10 device I use for my own work.</p>
<p>For cloning private the error message was</p>
<blockquote>
<p>Cloning into blah, blah…<br>
The program is blocked by group policy. For more information, contact your system administrator<br>
error: unable to read askpass response from ‘C:\Users\AppData\Local\GitHubDesktop\app-2.6.2\resources\app\static\ask-pass-trampoline.bat’<br>
fatal: could not read Username for ‘<a href="https://github.com" rel="noopener nofollow ugc">https://github.com</a>’ terminal prompts disabled</p>
</blockquote>
<p>A couple of options for you are<br>
a) Launch GitHub Desktop as administrator (right click, select ‘More’, select ‘Run as administrator’), O found this worked in my example<br>
b) Clone using SSH ( I suspect will work), although that will require additional configuration and setup to use</p>
<p>The underlying problem here is that GitHub Desktop internally executing a ‘bat’ command, as this is a potential attack vector, but that’s  separate discussion.</p>
<p>Let us know if this solves works around your problem or not.<br>
Good luck</p>

**Comment by Gummybear405** [comment]
<p>Oh my gosh, thank you! I ran the application as an administrator and pushed it again. It worked! Thank you so much for your help. :slight_smile:</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/gummybear405">@Gummybear405</a> great news :), if you mark the answer as solved in the post, it will show better up for anyone looking who has the same problem.<br>
Good luck with the rest of you work.</p>
